fty-shm-cleanup/src/fty_shm_cleanup.cc

```diff
@@ -50,108 +48,126 @@ static int parse_ttl(char* ttl_str, time_t& ttl)
     return 0;
 }
 
-// -1 : not a valid metric/data
+// -2 : file deletion failed
+// -1 : invalid file or metric/data
 //  0 : outdated data (file removed)
 //  1 : up to date data
-static int clean_outdated_data(std::string filename) {
-  FILE* file = fopen(filename.c_str(), "r");
+static int clean_outdated_data(std::string filename)
+{
+    FILE* file = fopen(filename.c_str(), "r");
+    if (!file) {
+        log_error("open %s failed (%s)", filename.c_str(), strerror(errno));
+        return -1;
+    }
 
-  if (!file) {
-    log_error("Cannot open %s for cleaning", filename.c_str());
-    return -1;
-  }
+    struct stat st;
+    if(fstat(fileno(file), &st) < 0) {
+        fclose(file);
+        log_error("stat %s failed (%s)", filename.c_str(), strerror(errno));
+        return -1; // invalid file
+    }
 
-  struct stat st;
-  if(fstat(fileno(file), &st) < 0) {
+    // read file in buf
+    char buf[128] = "";
+    fgets(buf, sizeof(buf), file);
     fclose(file);
-    return -1;
-  }
-
-  //read file in buf
-  char buf[128] = "";
-  fgets(buf, sizeof(buf), file);
-  fclose(file);
-  file = nullptr;
+    file = nullptr;
 
   //get ttl
   time_t ttl = -1;
   if (parse_ttl(buf, ttl) < 0) {
     return -1;
   }
-
+  
   //data still valid ?
   if (ttl >= 0) {
         time_t now = time(nullptr);
         if ((now - st.st_mtime) > ttl) {
             errno = ESTALE;
-            remove(filename.c_str());
-            return 0;
+            if (remove(filename.c_str()) != 0) {
+                log_error("remove %s failed (%s)", filename.c_str(), strerror(errno));
+                return -2; // rm failed
```

Comment:
why to return different error codes (btw, they are strange: -2, -1, 1, 0), if it will be ignored later?
Could this function just returns bool?